BatteryManager.cs

```diff
@@ -0,0 +1,150 @@
+﻿using System;
+using System.Globalization;
+using System.Resources;
+namespace BatteryManagementSystem 
+{
+    public class BatteryManager 
+    {
+        IReporter reporter;
+
+        public BatteryManager(IReporter reporterDependency)
+        {
+            if(reporterDependency == null)
+            {
+                throw new InvalidProgramException("Invalid program!!");
+            }
+            reporter = reporterDependency;
+        }
+
+        private static string language = "English";
+        public static string Language
+        {
+            get { return language; }
+            set { language = value; }
+        }
+        private static CultureInfo cultureInfo;
+        private static ResourceManager resourceManager = null;
+        public bool IsBatteryConditionOk(float temperature, float stateOfCharge, float chargeRate)
+        {
+            return IsTemperatureValid(temperature) && IsStateOfChargeValid(stateOfCharge) && IsChargeRateValid(chargeRate);
+        }
+        public bool IsTemperatureValid(float temperature)
+        {
+            bool isValid = new TemperatureValidator().IsValid(temperature);
+            DisplayMessageFromResourceFile(nameof(ResourceEnglish.TemperatueOutOfRange));
+            return isValid;
+        }
+        public bool IsStateOfChargeValid(float stateOfCharge)
+        {
+            bool isValid = new StateOfChargeValidator().IsValid(stateOfCharge);
+            DisplayMessageFromResourceFile(nameof(ResourceEnglish.StateOfChargeOutOfRange));
+            return isValid;
+        }
+        public bool IsChargeRateValid(float chargeRate)
+        {
+            bool isValid = new ChargeRateValidator().IsValid(chargeRate);
+            DisplayMessageFromResourceFile(nameof(ResourceEnglish.ChargeRateOutOfRange));
+            return isValid;
+        }
+
+        public bool IsBatteryBreached(float temperature, float stateOfCharge, float chargeRate)
+        {
+            return IsTemperatureLevelBreached(temperature) || IsStateOfChargeLevelBreached(stateOfCharge) || IsChargeRateLevelBreached(chargeRate);
+        }
+        private void DisplayMessageFromResourceFile(string resourceKey)
+        {
+            if (Language.Equals("German"))
```

Comment:
If we need to support new language will existing code adopt ?

Reply:
Provided extensibility to support new language